src/_tutorials/dart-vm/cmdline.md

```diff
@@ -104,62 +106,64 @@ This tutorial goes into detail about this app in the following sections.
-          .transform(const LineSplitter())
-          <a href="#" class="dart-popover" data-toggle="popover" title="Get data from a stream" data-html="true" data-trigger="hover focus" data-content="Listen to a stream to get data when it becomes available.">.listen((line)</a> {
-            if (showLineNumbers) {
-              stdout.write('${lineNumber++} ');
-            }
-            stdout.writeln(line);
-          }).asFuture().catchError((_) =&gt; _handleError(path));
-    });
+      <a href="#" class="dart-popover" data-toggle="popover" title="Open the stream" data-html="true" data-trigger="hover focus" data-content="Use the File class to represent a file on the native file system.">Stream lines = new File(path)</a>
+          <a href="#" class="dart-popover" data-toggle="popover" title="Open a file for reading" data-html="true" data-trigger="hover focus" data-content="Read the content from the file asynchronously.">.openRead()</a>
+          <a href="#" class="dart-popover" data-toggle="popover" title="Data converters" data-html="true" data-trigger="hover focus" data-content="Convert data as it becomes available on the stream.">.transform(UTF8.decoder)</a>
+          .transform(const LineSplitter());
+      <a href="#" class="dart-popover" data-toggle="popover" title="Try-catch block" data-html="true" data-trigger="hover focus" data-content="Catch errors by placing the asynchronous code inside a try-catch block.">try</a> {
+        <a href="#" class="dart-popover" data-toggle="popover" title="Get data from the stream" data-html="true" data-trigger="hover focus" data-content="Use 'await for' to loop over the values as they become available on the stream.">await for (var line in lines)</a> {
+          if (showLineNumbers) {
+            stdout.write('${lineNumber++} ');
+          }
+          stdout.writeln(line);
+        }
+      } <a href="#" class="dart-popover" data-toggle="popover" title="Try-catch block" data-html="true" data-trigger="hover focus" data-content="Catch errors by placing the asynchronous code inside a try-catch block.">catch</a> (_) {
+        _handleError(path);
+      }
+    }
   }
 }
 
-_handleError(String path) {
-  <a href="#" class="dart-popover" data-toggle="popover" title="Test a path" data-html="true" data-trigger="hover focus" data-content="You can get information about the file system on which your program is running.">FileSystemEntity.isDirectory(path)</a>.then((isDir) {
-    if (isDir) {
-      print('error: $path is a directory');
+Future _handleError(String path) async {
+  if (await <a href="#" class="dart-popover" data-toggle="popover" title="Test the path" data-html="true" data-trigger="hover focus" data-content="You can get information about the file system on which your program is running.">FileSystemEntity.isDirectory(path)</a>) {
```

Comment:
Should you add a highlight for "await", as well? Perhaps saying that the error handling pauses until the path is available. This could be part of the next highlight (or separate).